### PR TITLE
add test for route request values

### PR DESCRIPTION
**Pull request by jonathanolsson**
Add test for the route "request values".

### CODE REVIEW

File: database/database.js [hunk 1]
```diff
@@ -114,14 +114,12 @@ module.exports = {
...
 		
-
-			// console.log("Trying to add source value: " + sourceValueJSON + " into collection: " + collectionName);
 		try {
 			var createdSourceValue = new SourceValueModel(sourceValueJSON);
 		
 			createdSourceValue.save().then(function (res) {
 
-				// console.log("Successfully added source value " + res + " into collection " + collectionName);
+			//console.log("Successfully added source value " + res + " into collection " + collectionName);
```
Comment (jolin1337) [inline database/database.js]: Why is this here? If it is not needed remove console.log and maybe add a comment instead.

File: routes/routes.js [hunk 2]
```diff
@@ -68,57 +68,75 @@ var appRouter = function(app, upload, http, logger) {
 		};
 		//Writes in the conolse what was required. 
 		logger.log('debug', options);
-
+		
 		//Require data from a specified source API, the data is stored in 'chunk'.
 		var require = http.request(options, (result) => {
-			logger.log('debug', 'STATUS: '+ result.statusCode);
-			logger.log('debug', 'HEADERS: '+ JSON.stringify(result.headers));
+			//logger.log('debug', 'STATUS: '+ result.statusCode);
```
Comment (jolin1337) [inline routes/routes.js]: Is this needed?

File: routes/routes.js [hunk 3]
```diff
@@ -68,57 +68,75 @@ var appRouter = function(app, upload, http, logger) {
...
+			//logger.log('debug', 'HEADERS: '+ JSON.stringify(result.headers));
```
Comment (jolin1337) [inline routes/routes.js]: Is this needed?

File: routes/routes.js [hunk 4]
```diff
@@ -68,57 +68,75 @@ var appRouter = function(app, upload, http, logger) {
...
+		//Message on timeout.
+		require.on('timeout', (e) => {
+			res.status(400).send({result : {error : 1, message : "Timeout occured!"}});
+		});
 
 		//Message on error.
 		require.on('error', (e) => {
-			logger.log('error', 'problem with request: ' + e.message);
-			res.status(400).end("error: " + e.message)
+			//logger.log('error', 'problem with request: ' + e.message);
```
Comment (jolin1337) [inline routes/routes.js]: Is this needed?

File: routes/routes.js [hunk 5]
```diff
@@ -68,57 +68,75 @@ var appRouter = function(app, upload, http, logger) {
...
+			if(!res._headerSent){
+				console.log(res._headerSent);
```
Comment (jolin1337) [inline routes/routes.js]: Remove this!

File: routes/routes.js [hunk 6]
```diff
@@ -68,57 +68,68 @@ var appRouter = function(app, upload, http, logger) {
...
-			result.setEncoding('utf8')
-
+			result.setEncoding('utf8');
 			var collectionName = (req.body.sourceURL + req.body.port + req.body.path);
 
 			//Format the collectionname so it will not contain any special characters and then log into 'debug'.
 			collectionName = collectionName.replace(/[^A-Za-z0-9]/g, '');
-			logger.log('debug', 'COLLECTION NAME1: ' +  collectionName);
+
+			var message  ="";
```
Comment (paby1400) [inline routes/routes.js]: Remove this
